exe/main.cc

```diff
@@ -62,21 +74,17 @@ int main(int argc, char *argv[]) {
     Configuration::Instance().setStartFunction(startAgent);
 
     HW::ParseResult parse_result;
-    std::string err_msg {};
 
     try {
         parse_result = Configuration::Instance().initialize(argc, argv);
     } catch (const HW::horsewhisperer_error& e) {
         // Failed to validate action argument or flag
-        err_msg = e.what();
-    } catch(const Configuration::Error& e) {
-        std::cout << "A configuration error has occurred:\n  ";
-        err_msg = e.what();
-    }
-
-    if (!err_msg.empty()) {
-        std::cout << err_msg << "\nCannot start pxp-agent" << std::endl;
+        std::cout << e.what() << "\nCannot start pxp-agent'n";
         return 1;
+    } catch(const Configuration::Error& e) {
```

Comment:
You will start the dumb thread after for any error thrown in configuration.cc.
Wouldn't it be better to start it only if we miss the config file?
You could do that by throwing a specific error.